nrich-registry-spring-boot-starter/README.md

```diff
@@ -1,3 +1,85 @@
+
+With Maven:
+
+```xml
+
+<dependency>
+    <groupId>net.croz.nrich</groupId>
+    <artifactId>nrich-registry-spring-boot-starter</artifactId>
+    <version>${nrich.version}</version>
+</dependency>
+
+```
+
+With Gradle:
+
+```groovy
+
+implementation "net.croz.nrich:nrich-registry-spring-boot-starter:${nrich.version}"
+
+```
+
+Note if using `nrich-bom` dependency versions should be omitted.
+
+### Configuration
+
+Configuration is done through a property file, available properties and descriptions are given bellow (all properties are prefixed with nrich.registry which is omitted for readability):
+
+| property                                    | description                                                                                                                | default value                  |
+|---------------------------------------------|----------------------------------------------------------------------------------------------------------------------------|--------------------------------|
+| default-read-only-property-list             | List of property names that should always be marked as readonly                                                            |                                |
+| default-converter-enabled                   | Whether default string to type converter used for converting strings to property values when searching registry is enabled | true                           |
+| registry-search.date-format-list            | List of date formats used by to convert string to date value                                                               | dd.MM.yyyy, yyyy-MM-dd'T'HH:mm |
+| registry-search.decimal-number-format-list  | List of decimal formats used by to convert string to decimal value                                                         | #0.00, #0,00                   |
+| registry-search.boolean-true-regex-pattern  | Regexp pattern that is used by to match boolean true values                                                                | ^(?i)\s*(true                  |yes|da)\s*$ |
+| registry-search.boolean-false-regex-pattern | Regexp pattern that is used by to match boolean false values                                                               | ^(?i)\s*(false                 |no|ne)\s*$ |
+
+The properties under registry-search are used when converting string received from client to property values that will be used to search registry (i.e. if a string is sent and the property searched is
+of type `Date` then that string is parsed to `Date` and if parsing succeeds added as restriction to query).
+
+The default configuration values in yaml format for easier modification are given bellow:
+
+```yaml
+
+nrich.registry:
+  default-read-only-property-list:
+  default-converter-enabled: true
+  registry-search:
+    date-format-list: dd.MM.yyyy, yyyy-MM-dd'T'HH:mm
+    decimal-number-format-list: #0.00, #0,00
+    boolean-true-regex-pattern: ^(?i)\s*(true|yes|da)\s*$
```

Comment:
Default value here and in table above is not the same. Same applies for false regex pattern.